### DISCUSSION TITLE
Installing the BigTreeTech Mini e3 v3 in Sovol SV06

### DISCUSSION THREAD

**Discussion by transistor1** [bassamanator/everything-sovol-sv06#14] [General]
Here are some notes on how I made this swap, for future reference.  I'm sorry I didn't document this as well as I should have; hopefully I can answer questions if there are any.

Everything you need to do is inside of the motherboard casing. No additional wires need to be run outside of the casing. I started from [this](https://forum.sovol3d.com/t/sv06-mit-skr-e3-mini-v3/1189) German-language post on the Sovol forums (thanks Google translate!). Also got some help from Rory Game on the Sovol Facebook group (post image below). Few things to be aware of:

 - there are 2 jumpers that need to be connected for sensorless homing
 - Besides the power terminal block, which is oriented vertically in the Facebook photo below, there are 3 other horizontally-oriented blocks on the Sovol board, but only 2 on the BTT board:
  - Connection labeled +/- on the Sovol (third terminal block from left along the contiguous 3 blocks on the bottom of the board) is the extruder hot end and goes to HE0,
 - Unmarked cable (2nd from left) is the bed heater and goes to HB on the BTT. 
 - Fan +/- (leftmost) is connected to a terminal block in the must be terminated with an XH2.54 connector. Unfortunately that's a DIY. I cut a cable that was already terminated and soldered it to the end of the +/- connector. 
 - The cable labeled "DET" does not get attached to the BTT board. 

I used the [BTT documentation](https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/blob/master/hardware/BTT%20SKR%20MINI%20E3%20V3.0/Hardware/BTT%20SKR%20MINI%20E3%20V3.0%20user%20manual.pdf) and the photos from the Sovol forum to ensure I had a correct match on the other connectors.  

![image](https://user-images.githubusercontent.com/5625360/223434290-3a54d564-9105-4ed6-bd6a-a3142e937626.jpeg)

**Comment by transistor1** [comment]
Here are some photos of the inside of my board, for reference. 

![image](https://user-images.githubusercontent.com/5625360/223439554-421e6903-49b3-41dd-9c6a-4f47827a407c.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439581-2d4a3efc-c2c1-483e-b448-4fcd137dab7f.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439605-35eb0094-61d4-4d0f-a7ae-50eac79371aa.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439625-f40a4e10-77a4-4c3e-a7df-e98ee73c66c2.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439649-abc7e8ec-85d2-4857-a98b-ea9fef155fad.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439670-89fd2c37-e184-4878-8c8d-7d317253cfe4.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439695-35d93001-cd1e-4eae-b95c-8c418bf6db3d.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439718-7e2e82b5-7c84-402b-bbcb-7be2eef2f024.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439752-e7f1aabf-eaec-4d6b-8be3-1fb5f135192a.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439774-798a38e6-2c84-4443-9af1-aa30a668fc12.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439801-15794c23-0da1-4290-ac94-46da4a386431.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439820-7361c3a0-692e-40fd-b60a-2b9ff75abc34.jpeg)![image](https://user-images.githubusercontent.com/5625360/223439834-89bcdc64-173f-4091-b838-7301200ee532.jpeg)

**Comment by ClearAirTurbulence3D** [comment]
It looks like you didn't have to modify the case. The EZ board version is slightly longer, so it will probably need a new case. 

The DET connector may be from the filament runout sensor connector (P7) on the hot end.

**Comment by bassamanator** [comment]
@ClearAirTurbulence3D The DET cable is infact the filament runout sensor from the hotend PCB. I've got it working over here on my machine, just not fully ready for release because many adjustments are required in the config.

**Reply by transistor1** [reply]
Dumb question, but does the sovol have a built-in filament runout sensor?

**Reply by Astrolava** [reply]
No.

**Reply by ericyeargan** [reply]
On my SV06 setup, I've plugged DET into the ESTOP port and am using it for a KlackEnder switch via PC15.

**Comment by bassamanator** [comment]
@transistor1 Steve I've got the very basics so far, will be adding more tomorrow. Hopefully with your help, we can have a clear diagram that users can rely on when doing this mod. Heck, I'm so tempted myself thanks to you already having done the hard work 😁

I'll be adding all the basic stuff tomorrow or the day after. Hopefully after that you can add the more complicated stuff, like the jumpers you mentioned, etc.

![skr-mini](https://user-images.githubusercontent.com/61985779/223649218-de613792-cdcc-42b9-9efb-29e2ac44a8bf.png)

**Reply by transistor1** [reply]
That's awesome! Looks great!! Great work. Glad to help if it means it spares someone else some pain. 

> I'm so tempted myself

There's not too much advantage to swapping your board unless you do [Rory Game's mod](https://www.printables.com/model/382629-sovol-sv06-skr-3-ez-mainboard-case-upgrade) to allow you to control the two z-steppers independently. The advantage to that is that it will use the inductive to automatically align your gantry. 

...Or, of course, if you destroyed your board by trying to plug in components while the power was still on 😁. PSA: turn the power off when you are reconnecting your extruder.

**Comment by bassamanator** [comment]
@transistor1 Quick question: I noticed in `printer-skr-mini-e3-v3.cfg` that you've commented out the `display` section. Is the display not working?

**Comment by transistor1** [comment]
Yes, I haven't had a chance to get the display working yet. It's definitely
possible, because Rory Game had it working with his Marlin 2.1 fork, but I
don't yet know the correct pinout.

On Wed, Mar 8, 2023 at 2:39 PM Bassam Husain ***@***.***>
wrote:

> @transistor1 <https://github.com/transistor1> Quick question: I noticed
> in printer-skr-mini-e3-v3.cfg that you've commented out the display
> section. Is the display not working?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bassamanator/everything-sovol-sv06/discussions/14#discussioncomment-5245664>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK5MEF4CCXXOM7NPYJTVPLW3DN7HANCNFSM6AAAAAAVSPUH2M>
> .
> You are receiving this because you were mentioned.Message ID:
> <bassamanator/everything-sovol-sv06/repo-discussions/14/comments/5245664@
> github.com>
>
-- 
Sent from my phone

**Comment by bassamanator** [comment]
@transistor1 Got it.

Just want to confirm that `K-fan` is the part cooling fan?

I'm not sure what the jumper highlighted in yellow is?

Please correct any errors that you see, and anything that I've missed. It can be rough, I'll clean in on my end. When you have time, of course. Thanks!

![skr-mini4](https://user-images.githubusercontent.com/61985779/223918096-8326eef1-2767-4ef9-b4d8-b7d71c9f5c02.png)

**Reply by transistor1** [reply]
> Just want to confirm that `K-fan` is the part cooling fan?

I think K-Fan is the side fan mounted on the extruder, and Fan +/- is the nozzle fan.

 After I had been working on the pins for a while, I discovered that the official BTT site had a [klipper config](https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/blob/master/firmware/V3.0/Klipper/SKR-mini-E3-V3.0-klipper.cfg) for the Ender 3, and I had merged many of those changes into the config I had been working on.

In their config, `[heater_fan nozzle_cooling_fan]` is PC7 and `[fan]` is PC6.  BTT's [pin diagram](https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/blob/master/hardware/BTT%20SKR%20MINI%20E3%20V3.0/Hardware/BTT%20E3%20SKR%20MINI%20V3.0_PIN.pdf) shows FAN0 as PC6, and FAN1 as PC7.  [K-FAN goes to FAN0, and FAN +/- goes to FAN1](https://global.discourse-cdn.com/standard14/uploads/sovolforum/original/1X/4aa6b1120d337a4f1dcc30081aabf6803ec14f3b.jpeg).  If I need to get into the motherboard again, I'll try and remember to verify that with a multimeter.

<img width="194" alt="image" src="https://user-images.githubusercontent.com/5625360/224355045-d2768cc5-4f78-43d8-96ea-028e739e217c.png">


> I'm not sure what the jumper highlighted in yellow is?
> 

From [docs](https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/blob/master/hardware/BTT%20SKR%20MINI%20E3%20V3.0/Hardware/BTT%20SKR%20MINI%20E3%20V3.0%20user%20manual.pdf):
<img width="597" alt="image" src="https://user-images.githubusercontent.com/5625360/224352834-54b1be65-decd-41b6-805b-bcc2a99024e0.png">

**Comment by bassamanator** [comment]
@transistor1 Just one last thing: did you pull the changes I made, and now the LCD is working, and does everything else seem fine? Thanks!

**Comment by transistor1** [comment]
I only copied the LCD section.... didn't look at other changes. Will look
at that when I get a chance.

On Fri, Mar 10, 2023 at 11:17 AM Bassam Husain ***@***.***>
wrote:

> @transistor1 <https://github.com/transistor1> Just one last thing: did
> you pull the changes I made, and now the LCD is working, and does
> everything else seem fine? Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bassamanator/everything-sovol-sv06/discussions/14#discussioncomment-5269540>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABK5MEFMKONLTEL56GRJZ63W3NH2PANCNFSM6AAAAAAVSPUH2M>
> .
> You are receiving this because you were mentioned.Message ID:
> <bassamanator/everything-sovol-sv06/repo-discussions/14/comments/5269540@
> github.com>
>
-- 
Sent from my phone

**Comment by bassamanator** [comment]
@transistor1 Quick question. Did you have to change the physical pin layout on the `probe` connector, the part that connects to the board? Or was it plug and play?

(There are 2 users on Reddit that are going to be taking on this mod, and I'm trying to help them, *through you of course* 😁)

**Reply by transistor1** [reply]
Do you mean the connector itself?  No.  The only wiring change was the fan connector. 

By the way, you were correct- the K-Fan connection was the hot end fan and fan +/- was the side extruder cooling fan. 

I finally tried your whole Klipper profile but I couldn't get it to work. It was acting weird for me -  it kept homing in the wrong directions, so I reverted back to mine. I do not understand why that occurred. I did a diff of yours vs. mine and I didn't see any reversed pins in either 
profile or anything. None of the elements that were different seemed like they would warrant such weird behavior.   If anything,yours was more conservative with printing speeds and such

That being said, 2 days ago i since replaced my entire extruder with a new one I purchased and finally received from Sovol. It is now printing like a champ. I wonder if there was something wacky with the old extruder going on... though I don't know how that would lead to backward homing. 

I'm still using my profile because I'm scared to change anything. But I would be very interested to know how those other 2 people do with your config. I suspect it was a hardware issue on my side but would be nice to confirm.

**Reply by bassamanator** [reply]
So the probe sensor wiring does `not` need to be modified, and can be plugged into the motherboard as is, thanks! I'll let those other users know.

Your printer.cfg is the source of truth, as long as you haven't changed it, I will pull in your changes again. The reason why things might be a bit off is because I combined your printer.cfg with BTT official version found [here](https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3/blob/master/firmware/V3.0/Klipper/SKR-mini-E3-V3.0-klipper.cfg).

Appreciate your feedback. I'm pointing to your profile in the skr branch btw. Will keep you in the loop with the next changes I make. Thanks! 😀

**Reply by transistor1** [reply]
Don't pull in my changes, at least not yet.  I'm still a 3d-printing noob and still tinkering with things.  Your changes and knowledge have been spot-on so far (LCD screen, K-Fan, etc) so I don't want to consider it a config issue unless we can confirm.

**Reply by bassamanator** [reply]
I adjusted the x home direction. Also some improvements to the fan section.
